### PR TITLE
himalaya: 0.7.3 -> 0.8.0

### PR DESCRIPTION
**Pull request by soywod**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@toastal

### CODE REVIEW

**Comment by yanganto** [comment]
I am not sure it is a requirement or not, it may be good to squash these commits into one commit `himalaya: 0.7.3 -> 0.8.0`.

**Comment by soywod** [comment]
Done :heavy_check_mark:

**Comment by toastal** [comment]
Code looks okay. Builds as well.

```sh-session
$ nix build github:soywod/nixpkgs/himalaya-v0.8.0#himalaya
$ result/bin/himalaya --version
himalaya 0.8.0
```

**Comment by yanganto** [comment]
@soywod Some commits are mistaken into this PR

File: pkgs/applications/networking/mailreaders/himalaya/default.nix [hunk 1]
```diff
@@ -5,47 +5,31 @@
...
   pname = "himalaya";
-  version = "0.7.3";
+  version = "0.8.0";
 
   src = fetchFromGitHub {
     owner = "soywod";
     repo = pname;
     rev = "v${version}";
-    sha256 = "HmH4qL70ii8rS8OeUnUxsy9/wMx+f2SBd1AyRqlfKfc=";
+    sha256 = "kK/F3Geiuz0CgGSE0sCOfSZjg9gEPLyUEilzb+SdIM8=";
```
Comment (drupol) [inline pkgs/applications/networking/mailreaders/himalaya/default.nix:22]: Replace `sha256` with `hash`

**Review by drupol — CHANGES_REQUESTED** [review]
One small change and it's good to go.

**Comment by soywod** [comment]
> @soywod Some commits are mistaken into this PR

Which ones?

File: pkgs/applications/networking/mailreaders/himalaya/default.nix (hunk 1 above)
Comment (soywod) [reply pkgs/applications/networking/mailreaders/himalaya/default.nix]: Just by curiosity, what does it change? I see a lot of derivations with `sha256`.

Done :heavy_check_mark:

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
@grahamofborg build himalaya

**Comment by yanganto** [comment]
> @soywod Some commits are mistaken into this PR

It may not be a bug from github or not.  I saw a lot of changes on nixos modules and maintainers list at that time.

**Pull request merged**